src/qt3utils/applications/qt3scan/controller.py

```diff
@@ -200,7 +211,327 @@ def scan_image_rightclick_event(self, event: MouseEvent) -> None:
         """
         self.logger.debug(f"scan_image_rightclick_event. click at {event.xdata}, {event.ydata}")
 
-# class QT3ScanHyperSpectralApplicationController:
-#     """
-#     Implements qt3utils.applications.qt3scan.interface.QT3ScanApplicationControllerInterface
-#     """
\ No newline at end of file
+class QT3ScanHyperSpectralApplicationController:
+    """
+     Implements qt3utils.applications.qt3scan.interface.QT3ScanApplicationControllerInterface
+
+     For HyperSpectral imaging, the daq_controller object will be a spectrometer that
+     acquires a spectrum at each position in the scan.
+
+     Note that the DAQ controller here must implement QT3ScanCounterDAQControllerInterface
+
+
+     The following methods are required for a daq_controller object that
+     are not currently in the QT3ScanDAQControllerInterface.
+     We should find a way to declare these methods for any spectrometer
+     that is used for hyperspectral imaging. This could be another interface,
+     or perhaps using the concept of Python MixIns. To be discussed....
+
+    """
+    def __init__(self,
+                 position_controller: QT3ScanPositionControllerInterface,
+                 daq_controller: QT3ScanSpectrometerDAQControllerInterface,
+                 logger_level: int) -> None:
+
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logger_level)
+
+        self._daq_controller = daq_controller
+        self._position_controller = position_controller
+
+        self.running = False
+        self._current_y = 0
+        self._ymin = self.position_controller.minimum_allowed_position
+        self._ymax = self.position_controller.maximum_allowed_position
+        self._xmin = self.position_controller.minimum_allowed_position
+        self._xmax = self.position_controller.maximum_allowed_position
+
+        self._step_size = 0.5
+        self.raster_line_pause = 0.150  # wait 150ms for the piezo stage to settle before a line scan
```

Comment:
Does the `self.raster_line_pause` need to be set? What would be the use in someone adjusting this? As for the `self._step_size` does this not get overridden with the `step_size` value you set on the GUI?

Reply:
I think I only meant to talk about the raster line pause (yes, the step size is controllable). The raster line pause could be a necessary adjustment depending on the performance of a user's position controller. After thinking about it a bit, this might be something that has to go into the position controller configuration. Then the application controller would access this value via that object. This would allow a user to control that value...  I can probably write an issue around this. Let's just leave this code as is for now in this PR and we'll deal with the tech debt later in a separate chunk of work.

Reply:
I created an issue: https://github.com/qt3uw/qt3-utils/issues/141